Update CircleCI Config

calebcartwright opened the pull request:
* Adds test summary, closes https://github.com/badges/shields/issues/2635
* Removes generation of legacy secrets file (`private/secret.json`)
* Updates trigger to run twice daily and on commit

calebcartwright committed:
ci: updated circle to reflect shields config changes

calebcartwright commented on .circleci/config.yml:
```diff
@@ -49,5 +57,10 @@ workflows:
           filters:
             branches:
               only: master
+      - schedule:
```
I've never done multiple scheduled triggers in Circle before, and couldn't find anything decisive in their docs nor a quick google consultation. _Think_ this should work but if anyone knows otherwise (have to define a second workflow, can use multiple or better cron under single schedule key, etc.) please let me know!

paulmelnikow replied:
I think you can do it with cron!

https://crontab.guru/every-12-hours 😉

calebcartwright replied:
I wasn't sure if we wanted to maintain the original schedule time but yeah that will definitely be simpler!

paulmelnikow replied:
You could also do `45 3,15 * * *`.

paulmelnikow commented on .circleci/config.yml:
```diff
@@ -19,18 +19,23 @@ jobs:
-            echo "{\"gh_token\":\"$GITHUB_TOKEN\"}" > shields/private/secret.json
+          name: Prepare test summary
+          when: always
+          command: mkdir junit
```
I bet I added this to the main repo, though I wonder if it's needed. It seems to create at least one level of directory on its own (i.e. we're not creating `junit/app`).

I wonder if we should edit this so it triggers on pushes to PRs, while we're developing it.

calebcartwright replied:
I was wondering about that too. I know Mocha's xunit reporter will automatically create the directory hierarchy but didn't know about this reporter. I'll remove it to test

> I wonder if we should edit this so it triggers on pushes to PRs, while we're developing it.

Definitely!

paulmelnikow replied:
Pretty sure at this point, that this `mkdir` is a no-op 😁

calebcartwright committed:
ci: simplified cron trigger and added on-commit

calebcartwright committed:
ci: better cron, part 3 :)

calebcartwright committed:
ci: fixed typo in circle config

calebcartwright committed:
ci: temporarily removed branch filter for triggered workflow

calebcartwright committed:
ci: added script to config to debug missing test summary

calebcartwright committed:
ci: updated debug script to run always

paulmelnikow commented:
I suspect the branch filter is preventing this from running in the PR.

calebcartwright committed:
ci: added back test summary prep step

calebcartwright commented:
> I suspect the branch filter is preventing this from running in the PR.

I removed the branch filter temporarily 😄 It looks like we do need the setup step to create the junit directory. I've added that back, will see if it works in https://circleci.com/gh/badges/daily-tests/66 🤞

calebcartwright committed:
ci: still trying to debug missing tests

paulmelnikow commented on .circleci/config.yml:
```diff
@@ -46,7 +46,7 @@ jobs:
       - run:
           name: debugging missing tests
-          command: ls -R junit
+          command: ls -R shields/junit
```
👍

calebcartwright replied:
I just had a major facepalm moment 🤦‍♂️, really hope this is all it was the whole time

calebcartwright committed:
ci: updated test results path with correct directory

calebcartwright commented:
😌 this is finally working (https://circleci.com/gh/badges/daily-tests/68), and our coverity tests are consistently failing. 

do we want to add the master branch filter back to the on-commit trigger?

paulmelnikow commented:
> do we want to add the master branch filter back to the on-commit trigger?

What's your thought? I'd be inclined to leave it in the file commented out, until next time we're iterating on a PR.

paulmelnikow commented:
😍 

<img width="1021" alt="screen shot 2019-01-07 at 9 15 35 pm" src="https://user-images.githubusercontent.com/1487036/50805650-64c30000-12c1-11e9-892c-9230e9c5dc4c.png">

calebcartwright committed:
ci: finished updating circle config

calebcartwright commented:
I think we should leave it in, because as I personally found, sometimes you just need to run it on a PR. 

I added it back in (commented out as suggested) and removed the setup step that was creating the (definitely unneeded) junit directory

calebcartwright commented:
I know this is OT here but just to note, the entire coverity scan site appears to be inaccessible: https://scan.coverity.com 

I got an invalid cert error, and now getting redirected to the synopsys community page 🤔 I wonder if they're having issues or have decommissioned it like they did with their on-demand service (I added some details on that in https://github.com/badges/shields/pull/2550)

paulmelnikow commented:
> I think we should leave it in, because as I personally found, sometimes you just need to run it on a PR.

Do you mean an unmerged PR of the main repo? I just opened badges/shields#2685 for that, and was thinking of a different implementation. We could do it here, but it would require changing the git clone location and it seems a bit friendlier to do it in the PR title. I'm flexible though. Thoughts?

calebcartwright commented:
No I meant that IMO it is helpful for the daily-tests job to trigger on changes to this daily-tests repo, as the only time we are making changes within this repo is when we are trying to update/debug the daily-tests job configuration (or so I understand)

paulmelnikow commented:
I agree, though I think we need to see it before merge.

I see; if we remove the branch filter it'll run both before _and_ after, with no changes. I'm fine with that.

calebcartwright commented:
> I see; if we remove the branch filter it'll run both before _and_ after, with no changes. I'm fine with that.

Yup, exactly!
